### ISSUE TITLE
Unexpected token when when requiring pug files from required js files

### ISSUE THREAD

**Issue by bajax** [sidorares/pugify#38]
When I try to require pug files from the main javascript file-- as in, the one that gets passed directly to browserify-- pugify works flawlessly.

However, when I try to require a .js file that in turn requires a pug file, it fails with the 'Unexpected token' error, just like I'd attempted to include pug without pugify active.  

What gives?

EDIT: I have no idea what changed, but it's working now.  To be clear, the build was failing with just "Unexpected token"-- no further messages, no line numbers, nada.  The only common factor was including a pug template from a JS file, that was also included from somewhere else.

**Comment by bajax** [comment]
Nope, the problem remains. From the CLI, I can make it say that the error is on the very first line of the template file, pretty much as soon as it sees anything not javascript.

**Comment by bajax** [comment]
The weirdest thing...  I can make my JS files build by blanking out the content of the .pug file, trying to load it (it'll succeed because there are no unexpected tokens in an empty file) and then, when I try to generate my JS again after restoring the pug file's contents, it works.  There is no reason for this.  I'm using npm5, I've deleted and reinstalled my node_modules a few times, and I'm running nodejs 8.0.1.

Thanks.

**Comment by gwezerek** [comment]
I am also getting this error, though @bajax's fix isn't working.

**Comment by bajax** [comment]
@gwezerek The reason mine wasn't working was because I was requiring the pug templates through a symlink in node_modules.  Browserify transforms don't run on any file whose path is inside node_modules.  As for why it worked for me only inconsistently... search me.

**Comment by gwezerek** [comment]
Yes, thank you for updating and reminding me to! Indeed, I had the same problem. My solution was to add pugify (as a peer and dev dependency) and the pugify browserify transform to the package.json of the _linked_ module.
